src/cryptsetup/cryptsetup-generator.c

```diff
@@ -290,7 +290,7 @@ static int create_disk(
                 "Documentation=man:crypttab(5) man:systemd-cryptsetup-generator(8) man:systemd-cryptsetup@.service(8)\n"
                 "SourcePath=%s\n"
                 "DefaultDependencies=no\n"
-                "Conflicts=umount.target\n"
+                "Conflicts=shutdown.target\n"
```

Comment:
not sure i grok this change. I mean, doing "systemctl start umount.target" means all mounts are taken down, and doing "systemctl start shutdown.target" means all services are taken down. While taking down all mounts kinda implies taking down most services the opposite is not true: i.e. taking down all services shouldn't mean that all mounts are taken down too. Now, with your change you suddenly make it so that taking down all services also means detaching the storage behind mounts without necessarily taking down the mounts, and that doesn't really make sense, does it?

I mean, I think in the end these two targets are nor as independent as one might like, so the discussion is moot, but it's not clear to me why to have this commit hence. Can you elaborate?

Reply:
I can see your point.

When I opted for this change I considered the following points:

- systemd-cryptsetup@*.service are services so they should be stopped by shutdown.target in theory. But as you said it doesn't really make sense since stopping all services doesn't necessarily mean unmounting all filesystems.

- IMU, doing "systemctl start umount.target" is supposed to deal with mount units only, it isn't supposed to stop systemd-cryptsetup@.service and detach the disks behind mounts. Unmounting doesn't mean that the disk behind the mount should be removed, ejected or detached usually, does it ?

I'll drop this patch, however it still seems to me that conflicting with umount.target isn't correct.

Maybe the cryptsetup.target should have its counterpart target for shutting down crypto devices like umount.target is for mount.target.